OSDOCS-4347: Account Information: installing-aws-localzone created from cloudformation (#54535)
openshift/openshift-docs: modules/installation-creating-aws-subnet-localzone.adoc

// Module included in the following assemblies:
//
// * installing/installing_aws/installing-aws-localzone.adoc

:_content-type: PROCEDURE
[id="installation-creating-aws-subnet-localzone_{context}"]

> = Creating a subnet that uses local zones in AWS

Reviewer: ```suggestion
= Creating a subnet that uses AWS Local Zones
```

Author: Done, Thank You!

You must create a subnet in Amazon Web Services (AWS) that uses Local Zones for your {product-title}

> cluster to use.

Reviewer 2: I'm making a few grammar edits, so please confirm that the meaning is still correct.

have to contact Red Hat support with your installation logs.
====

.Prerequisites

* You configured an AWS account.
* You added your AWS keys and region to your local AWS profile by running `aws configure`.
* You opted in to the Local Zone group.

.Procedure

. Create a JSON file that contains the parameter values that the template
requires:
+
[source,json]
----
[
  {
    "ParameterKey": "ClusterName", <1>
    "ParameterValue": "mycluster" <2>
  },
  {
    "ParameterKey": "VpcId", <3>
    "ParameterValue": "vpc-<random_string>" <4>
  },
  {
    "ParameterKey": "PublicRouteTableId", <5>
    "ParameterValue": "<vpc_rtb_pub>" <6>
  },
  {
    "ParameterKey": "LocalZoneName", <7>
    "ParameterValue": "<cluster_region_name>-<location_identifier>-<zone_identifier>" <8>
  },
  {
    "ParameterKey": "LocalZoneNameShort", <9>
    "ParameterValue": "<lz_zone_shortname>" <10>
  },
  {
    "ParameterKey": "PublicSubnetCidr", <11>
    "ParameterValue": "10.0.128.0/20" <12>
  }
]
----
<1> A short, representative cluster name to use for hostnames, etc.
<2> Specify the cluster name that you used when you generated the `install-config.yaml` file for the cluster.

> <3> The VPC created resources will belong to.

Reviewer 2: I blame the [prior art](https://github.com/openshift/openshift-docs/blame/main/modules/installation-creating-aws-bootstrap.adoc#L144). If we get more details, it might be worth fixing both instances.

> <3> The VPC created resources will belong to.

Reviewer: ```suggestion
<3> The VPC ID that the subnet will be created.
```

Reviewer 2: @mtulio, do you mean something like "The VPC ID in which the cluster's subnet will be created?"

Reviewer: @kalexand-rh the subnet is part(belongs to) of VPC, so I think something like this could be better: "The VPC ID in which the Local Zone's subnet will be created".

<4> Specify the `VpcId` value from the output of the CloudFormation template
for the VPC.
<5> The Public Route Table ID for the VPC.
<6> Specify the `PublicRouteTableId` value from the output of the CloudFormation template for the VPC.
<7>  The Local Zone name that the VPC belongs to.
<8> Specify the Local Zone that you opted your AWS account into, such as `us-east-1-nyc-1a`.
<9> The shortname of the Local Zone that the VPC belongs to.
<10> Specify the Local Zone that you opted your AWS account into, such as `nyc1`.
//How do we determine this shortname?
<11> The CIDR block to allow access to the Local Zone.
<12> Specify a CIDR block in the format `x.x.x.x/16-24`.
//How do we know what this CIDR is?

> . Copy the template from the *CloudFormation template for the VPC*

Reviewer 3: Should be `CloudFormation template for the subnet`?

section of this topic and save it as a YAML file on your computer. This template
describes the VPC that your cluster requires.

. Launch the CloudFormation template to create a stack of AWS resources that represent the VPC:
+
[IMPORTANT]
====
You must enter the command on a single line.
====
+
[source,terminal]
----
$ aws cloudformation create-stack --stack-name <name> <1>
     --template-body file://<template>.yaml <2>

>      --parameters file://<parameters>.json <3>

Reviewer 2: The previous note indicates that these must all be entered on a single line. Breaking a multi-line command like this has made users open bugs before because of rendering issues, IIRC.

> $ aws cloudformation create-stack --stack-name <name> \ <1>

Reviewer: Suggest to change to:
```
$ aws cloudformation create-stack --stack-name <subnet_stack_name> \ <1>
```

[1] Section: `Creating the Kubernetes manifest files`
Procedure.2: `Create the machine set manifests for the worker nodes in your Local Zone.`
Step: `Store the subnet ID as a local variable by running the following command:`

----
<1> `<name>` is the name for the CloudFormation stack, such as `cluster-lz-<local_zone_shortname>`.
You need the name of this stack if you remove the cluster.
<2> `<template>` is the relative path to and name of the CloudFormation template
YAML file that you saved.
<3> `<parameters>` is the relative path to and name of the CloudFormation
parameters JSON file.
+
.Example output
[source,terminal]
----
arn:aws:cloudformation:us-east-1:123456789012:stack/cluster-lz-nyc1/dbedae40-2fd3-11eb-820e-12a48460849f
----

> . Confirm that the template components exist:

Reviewer 2: I shall fix it, but I blame [this](https://github.com/openshift/openshift-docs/blame/main/modules/installation-creating-aws-bootstrap.adoc#L204). We codified that style guidance after, and I missed it in the copied file.
